### PR TITLE
feature(VILLB-300-301): create abstract file storage service and s3fileStorage service.

### PR DESCRIPTION
**Pull request by katafalque**

### CODE REVIEW

File: src/providers/aws/s3-file-storage.config.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export class S3FileStorageConfig{
+    s3ServiceAccessKey: string;
+    s3ServiceSecretKey: string;
+    s3ServiceRegion: string;
+    s3ServiceBucket: string;
```
Comment (erailea) [inline src/providers/aws/s3-file-storage.config.ts]: it is already s3config so no need for redundancy.

accessKey
secretKey
region
bucket

**Review by erailea — COMMENTED** [review]
lint and format required.

File: src/file-storage.module.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            provide: "S3FileStorageConfig",
+            useFactory: (configService: ConfigService) => configService.get("aws"),
+            inject: [ConfigService],
+        },
+        S3FileStorageService,
+        {
+            provide: 'S3',
+            useFactory: () => {
+                return new S3({
+                    accessKeyId: '',
```
Comment (erailea) [inline src/file-storage.module.ts]: should get from config

File: src/file-storage.module.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { Module } from "@nestjs/common";
+import { ConfigService } from '@nestjs/config';
+import { S3 } from "aws-sdk";
+import { S3FileStorageService } from "./providers/aws/s3-file-storage.service";
+import { FileStorageType } from "./enum/file-storage-type.enum";
+
+
+@Module({
+    imports: [],
```
Comment (erailea) [inline src/file-storage.module.ts]: should import ConfigModule with storage config

**Review by erailea — CHANGES_REQUESTED** [review]

File: src/file-storage.service.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export abstract class FileStorageService {
+  abstract store(file: Buffer, path: string, config: any);
```
Comment (ismetkizgin) [inline src/file-storage.service.ts]: Why don't we say upload instead of store?

Comment (ismetkizgin) [inline src/file-storage.service.ts]: Return type must be specified.

File: src/file-storage.service.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+  abstract delete(path: string);
```
Comment (ismetkizgin) [inline src/file-storage.service.ts]: Return type must be specified.

**Review by ismetkizgin — CHANGES_REQUESTED** [review]

File: src/file-storage.module.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      inject: [ConfigService],
+    },
+    {
+      provide: 'FileStorageService',
+      useFactory: (
+        s3FileStorageService: S3FileStorageService,
+        configService: ConfigService,
+      ) => {
+        const fileStorageType = configService.get('storageService');
+        console.log(fileStorageType);
```
Comment (erailea) [inline src/file-storage.module.ts]: remove console log

File: src/file-storage.module.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        switch (fileStorageType) {
+          case FileStorageType.S3:
+            return s3FileStorageService;
+          default:
+            throw new Error('Wrong Storage Type');
```
Comment (erailea) [inline src/file-storage.module.ts:45]: add test for invalid storage type

File: src/file-storage.module.spec.ts [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        FileStorageModule,
+      ],
+    }).compile();
+
+    fileStorageModule = app.get<FileStorageModule>(FileStorageModule);
+    expect(fileStorageModule).toBeDefined();
+  });
+
+  it('Should be defined with env configuration', async () => {
+    process.env.STORAGE_SERVICE = FileStorageType.S3;
```
Comment (ismetkizgin) [inline src/file-storage.module.spec.ts:34]: Env should not be mocked directly, it may affect the tests. ConfigService should be mocked directly.

**Review by erailea — CHANGES_REQUESTED** [review]
add main to package.json

**Review by erailea — CHANGES_REQUESTED** [review]
remove private true from package.json

**Review by erailea — CHANGES_REQUESTED** [review]
add readme

**Review by erailea — CHANGES_REQUESTED** [review]
add release library and script

**Review by erailea — CHANGES_REQUESTED** [review]
move @nestjs libraries to version 9 to keep a standart between @brewww libraries

**Review by erailea — CHANGES_REQUESTED** [review]
remove unused libraries like automapper

**Review by erailea — CHANGES_REQUESTED** [review]
add another store method for upload with base64.

**Review by mehmetsagdur — CHANGES_REQUESTED** [review]
Project name must start with @brewww/ for package.json

**Comment by erailea** [comment]
remove e2e tests and configs

**Pull request merged**